src/serac/numerics/assembled_sparse_matrix.cpp

```diff
@@ -0,0 +1,274 @@
+
+#include "mfem/linalg/dtensor.hpp"
+#include "assembled_sparse_matrix.hpp"
+
+template <auto offsetV, auto indicesV, auto gatherV>
+struct forbidden_restriction {
+  //  friend result_fes _get_fes(From & from) { return & from.*fesV; }
+  friend mfem::Array<int>& __get_offsets(mfem::ElementRestriction& from) { return from.*offsetV; }
+  friend mfem::Array<int>& __get_indices(mfem::ElementRestriction& from) { return from.*indicesV; }
+  friend mfem::Array<int>& __get_gatherMap(mfem::ElementRestriction& from) { return from.*gatherV; }
+};
```

Comment:
I presume this is related to getting at protected members of `mfem::ElementRestriction`, but this needs some explanation.

Reply:
Here's an article that explains the trick being used: https://accu.org/journals/overload/28/156/harrison_2776/

In addition to linking this we could also document why it's needed for this particular case and maybe raise an MFEM issue since presumably this isn't the long-term solution we want